model/flow/sealing_segment.go

```diff
@@ -16,22 +16,51 @@ const (
 // height) is this snapshot's reference block and the first (least height)
 // is the most recently sealed block as of this snapshot (ie. the block
 // referenced by LatestSeal).
-// In other words, the last block contains a seal for the first block.
-// For instance:
-//   A <- B <- C <- D <- E (seal_A)
+//
+// In other words, the most recently incorporated seal as of the highest block
+// references the lowest block. The highest block does not need to contain this
+// seal.
+//
+// Example 1 - E seals A:
+//   A <- B <- C <- D <- E(SA)
 // The above sealing segment's last block (E) has a seal for block A, which is
 // the first block of the sealing segment.
 //
-// It is guaranteed there are at least 2 blocks in a SealingSegment
+// Example 2 - E contains no seals, but latest seal prior to E seals A:
+//   A <- B <- C <- D(SA) <- E
+//
+// It is guaranteed there are at least 2 blocks in a SealingSegment, unless
```

Comment:
I am not sure this statement is correct, our seal validator and builder use new rules, to be more specific we require a gap between incorporated result and seal, we can't construct something like:
`A <- B[ER_A] <- C[S_A]` we actually need to have a chain: `A <- B[ER_A] <- C <- D[S_A]`

Reply:
It's a technical requirement for having correct SoR, more here: https://github.com/onflow/flow-go/blob/d69930a2472c99c7e51e58155a0de46022b267e7/module/validation/seal_validator.go#L162